mapflow/mapflow.py

```diff
@@ -1539,10 +1539,10 @@ def calculate_aoi_area_polygon_layer(self, layer: Union[QgsVectorLayer, None]) -
             return
 
         features = list(layer.getSelectedFeatures()) or list(layer.getFeatures())
-        if layer.wkbType() == QgsWkbTypes.MultiPolygon:
-            geoms_count = layer_utils.count_polygons_in_layer(features)
-        elif layer.wkbType() == QgsWkbTypes.Polygon:
+        if layer.wkbType() == QgsWkbTypes.Polygon or QgsWkbTypes.PolygonZ or QgsWkbTypes.PolygonZM:
```

Comment:
"or" does not work this way, here it would apply to the condition  (cast 2nd and 3rd arguments to bool and compare).

Change to `layer.wkbType() in (....)`